.github/workflows/docker-connector-image-publishing.yml

```diff
@@ -0,0 +1,108 @@
+      image-type:
+        description: "Image Type (for now only 'base' is supported)"
+        required: true
+        default: 'base'
+        type: choice
+        options:
+          - 'base'
+      tag-or-version-number:
+        description: "Tag or Version Number (without 'v' prefix)"
+        required: true
+        default: 'dev-test'
+      repository-root:
+        description: "Repository Root, either GitHub (GHCR) or DockerHub. Use GHCR for testing."
+        required: true
+        default: 'ghcr.io/airbytehq'
+        type: choice
+        options:
+          - 'docker.io/airbyte'
+          - 'ghcr.io/airbytehq'
+      dry-run:
+        description: "Dry Run (if true, no image will be pushed)"
+        required: true
+        type: boolean
+        default: true
+      require-security-check:
+        description: "Abort if HIGH/CRITICAL security vulnerabilities are found."
+        required: true
+        type: boolean
+        default: true
+
+jobs:
+  build-and-publish-base-image:
+    name: Build and Publish ${{ github.event.inputs.connector-type }} Base Image
+    runs-on: ubuntu-latest
+    if: ${{ github.event.inputs.image-type == 'base' }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 1
+
+      - name: Log in to DockerHub
+        uses: docker/login-action@v3
+        if: ${{ github.event.inputs.repository-root == 'docker.io/airbyte' }}
+        with:
+          registry: ${{ github.event.inputs.repository-root }}
+          username: ${{ secrets.DOCKER_HUB_USERNAME }}
+          password: ${{ secrets.DOCKER_HUB_PASSWORD }}
+
+      - name: Log in to GHCR
+        uses: docker/login-action@v3
+        if: ${{ github.event.inputs.repository-root == 'ghcr.io/airbytehq' }}
+        with:
+          registry: ghcr.io/airbytehq
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Set image build variables
+        id: vars
+        run: |
+          DOCKERFILE=docker-images/Dockerfile.${{ github.event.inputs.connector-type }}-connector-base
+          IMAGE_NAME=${{ github.event.inputs.connector-type }}-connector-base
+          FULL_IMAGE_REF=${{ github.event.inputs.repository-root }}${IMAGE_NAME}:${{ github.event.inputs.tag-or-version-number }}
+          echo "DOCKERFILE=${DOCKERFILE}" | tee -a "$GITHUB_OUTPUT"
+          echo "IMAGE_NAME=${IMAGE_NAME}" | tee -a "$GITHUB_OUTPUT"
+          echo "FULL_IMAGE_REF=${FULL_IMAGE_REF}" | tee -a "$GITHUB_OUTPUT"
+
+      - name: Build Base Image
+        id: docker-build-base
+        uses: docker/build-push-action@v6
+        with:
+          file: ${{ steps.vars.outputs.DOCKERFILE }}
+          platforms: linux/amd64,linux/arm64
+          tags: ${{ steps.vars.outputs.FULL_IMAGE_REF }}
+          push: false
+
+      - name: Run Vulnerability Scan
```

Comment:
thought: this might block a base image that doesn't have existing fixes. would that be an issue?

Reply:
@davinchia - Yeah, I made it intentionally easy to skip - simply ru-running with `require-security-check==false` will make it informative, but not fatal.

Basically: this prevents _accidentally_ publishing an image with already-known High/Severe vulnerabilities, but it doesn't block _informed_ publishing of the same.